packages/xo-server/src/api/sr.js

```diff
@@ -816,6 +816,20 @@ export async function reattach({ host, uuid, nameLabel, nameDescription, type })
 
   const srRef = await xapi.call('SR.introduce', uuid, nameLabel, nameDescription, type, 'user', true, {})
 
+  if (deviceConfig !== undefined) {
+    const hosts = filter(xapi.objects.all, object => object.$type === 'host')
+    await Promise.all(
+      hosts.map(async host => {
+        const pbdRef = await xapi.call('PBD.create', {
+          host: host.$ref,
+          SR: srRef,
+          device_config: deviceConfig,
+        })
+        await xapi.call('PBD.plug', pbdRef)
+      })
+    )
+  }
```

Comment:
Should we do this for non-shared SRs?

Reply:
This is only done for network SRs so I think we should always do it. Plus I'm not sure how we'd  get any other info about which hosts to reconnect since we only get the SRs' UUIDs.